semantic_router/index/pinecone.py

```diff
@@ -866,22 +1031,29 @@ def _read_config(self, field: str, scope: str | None = None) -> ConfigParameter:
             ids=[config_id],
             namespace="sr_config",
         )
-        if config_record.get("vectors"):
-            return ConfigParameter(
-                field=field,
-                value=config_record["vectors"][config_id]["metadata"]["value"],
-                created_at=config_record["vectors"][config_id]["metadata"][
-                    "created_at"
-                ],
-                scope=scope,
+        # Pinecone v7: FetchResponse with .vectors mapping id -> Vector
+        if hasattr(config_record, "vectors") and config_id in config_record.vectors:
+            vec = config_record.vectors[config_id]
+            md = getattr(vec, "metadata", {}) or {}
+            value = md.get("value", "")
+            created_raw = md.get("created_at")
+            created_at: str = (
+                created_raw
+                if isinstance(created_raw, str)
+                else datetime.now(timezone.utc).isoformat()
             )
-        else:
-            logger.warning(f"Configuration for {field} parameter not found in index.")
             return ConfigParameter(
                 field=field,
-                value="",
+                value=value,
+                created_at=created_at,
                 scope=scope,
             )
```

Comment:
should rename `md` to `metadata` - it's confusing otherwise.

Also - why would `created_at` not be a `str` type? Unless there's a reason can we assume it is and otherwise raise a `TypeError`